### PR TITLE
Delete calendars with their users

### LINKED ISSUE #1551: Data of deleted user still there
### Steps to reproduce
1. Delete User via Admin-Panel
2. Create same User again via Admin Panel
3. Login at User account, loading Contacts App

### Expected behaviour
New user, there should not be any datas from the deleted user

### Actual behaviour
- Data from database (contacts) is still there
- Also, nextcloud asked for two factor auth for the new user. Two factor auth was activated at the deleted user. After disabling two factor auth, I could login.

### Server configuration
**Operating system**: Ubuntu 16.04

**Web server:** Apache 2.4

**Database:** 10.0.27-MariaDB

**PHP version:** PHP 7.0.8

**Nextcloud version:** 10.0

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** download from web

**Signing status:**
No errors have been found.

**List of activated apps:**
standard list
+
contacts
calendar
two factor auth 
mail
tasks

**The content of config/config.php:**
<details>
<summary>Config report</summary>

<?php
$CONFIG = array (
  'instanceid' => 'ocr3jwxw0325',
  'passwordsalt' => '....',
  'secret' => '....',
  'trusted_domains' => 
  array (
    0 => 'example.eu:444',
  ),
  'datadirectory' => '/mnt/nfs/webserver_daten/nextcloud_data',
  'overwrite.cli.url' => 'https://example.eu:444',
  'dbtype' => 'mysql',
  'version' => '9.1.0.16',
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'nextcloud',
  'dbpassword' => '.....',
  'logtimezone' => 'UTC',
  'installed' => true,
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'hosts' => 'localhost',
    'port' => 6379,
  ),
  'appstore.experimental.enabled' => false,
);
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 49.0.1

**Operating system:** Windows 10

No errors in logs

### PR DESCRIPTION
**Pull request by LukasReschke**
From https://github.com/owncloud/core/pull/25918

Fix #1551

### CODE REVIEW

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -508,6 +550,16 @@ function deleteCalendar($calendarId) {
 	}
 
 	/**
+	 * Delete all of an user's shares
+	 *
+	 * @param string $principaluri
+	 * @return void
+	 */
+	function deleteAllSharesForUser($principaluri) {
+		$this->sharingBackend->deleteAllSharesByUser($principaluri);
```
Comment (nickvergessen) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: For or By? This looks like a trap for the future 🙈

**Comment by nickvergessen** [comment]
Works 👍

**Comment by nickvergessen** [comment]
This also was reported yesterday: https://github.com/nextcloud/server/issues/1551

I'd like to see a backport to 10 for this.

**Comment by LukasReschke** [comment]
> This also was reported yesterday: #1551

I think that ticket is about contacts, not calendar data.

**Comment by nickvergessen** [comment]
See the second commit? in this PR?

File: apps/dav/lib/CalDAV/CalDavBackend.php (hunk 1 above)
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: Fixed, thanks.

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**